### PR TITLE
Expanding CRD definition, adding OpenAPIv3 schema

### LINKED ISSUE #17: Expand the spec of our Product CR + add openAPIV3Schema to CRD
The Product CRs need to include valuable information other than its description. They require to match, more or less, the 3scale Product API.
When creating the Products, we need to add the fields name, system_name, apps_count, last_updated, backends_count, description, account and unread_alerts
Also, these fields need to be described in the CRD with openAPIV3Schema

### PR DESCRIPTION
**Pull request by damianpm**
Closes https://github.com/3scale/k8sapp-initiative/issues/17

It expands our CRD with an OpenAPIv3 schema for products
It also expands our products properties to match current 3scale products

**VERIFICATION STEPS:**

Apply changes to CRD:
```
oc apply -f product_crd.yml 
```

Create products:
```
oc apply -f products_examples.yml
```
Verify products are created:
```
oc get products.k8sinitiative.3scale.net 
```
You shoould see something like:
```
NAME             AGE
product-api-01   24h
product-api-02   24h
```

Verify a single product in yaml format:
```
oc get products.k8sinitiative.3scale.net product-api-01 -o yaml 
```
You should see:
```
apiVersion: k8sinitiative.3scale.net/v1alpha1
kind: Product
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"k8sinitiative.3scale.net/v1alpha1","kind":"Product","metadata":{"annotations":{},"name":"product-api-01","namespace":"k8sinitiative"},"spec":{"apps_count":1,"backends_count":1,"description":"Product 1","id":1,"link":"/apiconfig/services/1","links":[{"name":"Edit","path":"/apiconfig/services/1/edit"},{"name":"Overview","path":"/apiconfig/services/1"},{"name":"Analytics","path":"/services/1/stats/usage"},{"name":"Applications","path":"/apiconfig/services/1/applications"},{"name":"ActiveDocs","path":"/apiconfig/services/1/api_docs"},{"name":"Integration","path":"/apiconfig/services/1/integration"}],"name":"API 1","system_name":"api_1","unread_alerts_count":0,"updated_at":"September 22, 2020 10:23"}}
  creationTimestamp: 2020-11-19T11:59:09Z
  generation: 3
  managedFields:
  - apiVersion: k8sinitiative.3scale.net/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        f:apps_count: {}
        f:backends_count: {}
        f:id: {}
        f:link: {}
        f:links: {}
        f:name: {}
        f:system_name: {}
        f:unread_alerts_count: {}
        f:updated_at: {}
    manager: kubectl-client-side-apply
    operation: Update
    time: 2020-11-20T11:25:18Z
  - apiVersion: k8sinitiative.3scale.net/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
      f:spec:
        .: {}
        f:description: {}
    manager: oc
    operation: Update
    time: 2020-11-20T11:44:48Z
  name: product-api-01
  namespace: k8sinitiative
  resourceVersion: "68150324"
  selfLink: /apis/k8sinitiative.3scale.net/v1alpha1/namespaces/k8sinitiative/products/product-api-01
  uid: 4664c8d4-a232-4143-8e95-f8a1bdc236db
spec:
  apps_count: 1
  backends_count: 1
  description: Product 1
  id: 1
  link: /apiconfig/services/1
  links:
  - name: Edit
    path: /apiconfig/services/1/edit
  - name: Overview
    path: /apiconfig/services/1
  - name: Analytics
    path: /services/1/stats/usage
  - name: Applications
    path: /apiconfig/services/1/applications
  - name: ActiveDocs
    path: /apiconfig/services/1/api_docs
  - name: Integration
    path: /apiconfig/services/1/integration
  name: API 1
  system_name: api_1
  unread_alerts_count: 0
  updated_at: September 22, 2020 10:23
```

### CODE REVIEW

File: products_examples.yml [hunk 1]
```diff
@@ -4,12 +4,55 @@ metadata:
...
+    - name: "Analytics"
+      path: "/services/2/stats/usage"
+    - name: "Applications"
+      path: "/apiconfig/services/2/applications"
+    - name: "ActiveDocs"
+      path: "/apiconfig/services/2/api_docs"
+    - name: "Integration"
+      path: "/apiconfig/services/2/integration"
+  unread_alerts_count: 0
+  updated_at: "September 22, 2020 10:23"
```
Comment (didierofrivia) [inline products_examples.yml]: This would need to be a timestamp

File: products_examples.yml [hunk 2]
```diff
@@ -4,12 +4,55 @@ metadata:
...
+
```
Comment (didierofrivia) [inline products_examples.yml]: don't forget to add a new line at the end of file ;)

File: product_crd.yml [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+                    type: integer
+                  id:
+                    type: integer
+                  name:
+                    type: string
+                  system_name:
+                    type: string
+                  link:
+                    type: string
+                  links:
```
Comment (didierofrivia) [inline product_crd.yml:32]: good to have them, not super necessary for the POC tho

Comment (didierofrivia) [reply product_crd.yml:32]: question: are you running a OAS3 validator?

**Review by didierofrivia — CHANGES_REQUESTED** [review]
most important would be the timestamp for updated and created at

File: product_crd.yml (hunk 3 above)
Comment (damianpm) [reply product_crd.yml:32]: I'm not running any validator, but when you run `kubectl apply` it validates the schema
Screenshot 2020-11-23 at 10 51 09

**Review by didierofrivia — APPROVED** [review]

**Pull request merged**